### PR TITLE
Us040 login system

### PR DESCRIPTION
**Pull request by steveh2707**

### CODE REVIEW

**Review by DarraghHaughey-Kainos — APPROVED** [review]
Login works for me!

File: app.ts [hunk 1]
```diff
@@ -25,10 +28,32 @@ app.use(express.json());
 
 app.use(express.urlencoded({ extended: true }));
 
+app.use(session({
+    secret: 'NOT HARDCODED SECRET',
```
Comment (glenn-hamilton) [inline app.ts]: `NOT HARDCODED SECRET`

File: controller/AuthController.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        } catch(e) {
+            res.locals.errormessage = e.message;
+
+            res.render("login", req.body)
+        }
+    })
+
+    app.get('/register', async (req: Request, res: Response) => {
+        
+        res.render('Register', {title: "Register"});
```
Comment (glenn-hamilton) [inline controller/AuthController.ts]: `Register` as the HTML template should be lower case

File: controller/HelloWorldController.ts [hunk 3]
```diff
@@ -1,22 +1,17 @@
 import { Application, Request, Response } from "express";
```
Comment (glenn-hamilton) [inline controller/HelloWorldController.ts]: You can remove all the usages of your `HelloWorld` test

File: controller/HomeController.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Application, Request, Response } from "express";
+
+
+module.exports = function(app: Application) {
+
+    app.get('/', async (req: Request, res: Response) => {
+
+        res.render('Home', {title: "Home"});
```
Comment (glenn-hamilton) [inline controller/HomeController.ts]: `Home` should be lower case

File: test/unit/service/AuthService.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+            mock.onPost(`${API_BASE_URL}/login`, login).reply(200, token)
+
+            const result: String = await authService.login(login);
+
+
+            expect(result).to.equal(token);
+        })
+
+        it('should throw expception when 500 error returned from axios', async () => {
```
Comment (glenn-hamilton) [inline test/unit/service/AuthService.ts]: `it('should throw exception when 500 error returned from axios', async () => {`

File: test/unit/service/AuthService.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    describe('register', function() {
+        it('should return token on successful registration', async () => {
+            const mock = new MockAdapter(axios);
+
+            mock.onPost(`${API_BASE_URL}/register`, login).reply(200, token)
+
+            expect(token).to.equal(token);
+        })
+
+        it('should throw expception when 500 error returned from axios', async () => {
```
Comment (glenn-hamilton) [inline test/unit/service/AuthService.ts]: `it('should throw exception when 500 error returned from axios', async () => {`

File: views/Login.html
Comment (glenn-hamilton) [inline views/Login.html:1]: Can you make the filename lower case

File: views/Register.html
Comment (glenn-hamilton) [inline views/Register.html:1]: Can you make the filename lower case

File: views/_includes/layouts/base.html [hunk 7]
```diff
@@ -19,8 +20,22 @@
...
                     <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
-                        <div class="navbar-nav">
-                            <a class="nav-item nav-link active" href="/">Home<span class="sr-only">(current)</span></a>
+                        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+                            <li class="nav-item">
+                                <a class="nav-link" href="#">Link 1</a>
+                            </li>
+                            <!-- Each item to be added list item here -->
+                        </ul>
+                        <div class="right-aligned">
```
Comment (glenn-hamilton) [inline views/_includes/layouts/base.html:34]: This causes an accessibility issue, as if CSS is turned off nothing appears here

**Review by glenn-hamilton — APPROVED** [review]

**Pull request merged**